### PR TITLE
doc: Remove protocol descriptions

### PR DESCRIPTION
**Pull request by secworks**
Remove 'interface_fpga' and try to replace with 'MCU'
     Try to fix wording to be in sync with MCU + one FPGA
     Remove chapter about FPGA-FPGA communication

### CODE REVIEW

**Comment by dehanj** [comment]
Updated to remove the framing protocol competly from the repo and point to dev.tillitis.se.

Probably even more documents needs to be removed, this is a start.

**Comment by mchack-work** [comment]
Also removed firmware protocol and point to dev handbook.

Rebased on main.

Now a real PR.

File: doc/system_description/software.md [hunk 1]
```diff
@@ -255,188 +256,6 @@ Then we continue with the CDI computation by updating with an optional
 USS and then finalizing the hash, storing the resulting digest in
 `CDI`.
 
-### Firmware protocol
```
Comment (dehanj) [inline doc/system_description/software.md:258]: With these changes, should we remove the note on the top of the document?
Or what do we foresee that our documentation should live?

Right now we have three; here, close to the cores and dev.tillitis. Should we keep all?

Comment (dehanj) [reply doc/system_description/software.md:258]: Might be a bigger discussion than this specific PR.

Comment (mchack-work) [reply doc/system_description/software.md:258]: I think the direction things are going according to earlier chats is that we move specific documentation closer to the thing it documents, so I think the firmware parts of `software.md` should move to a README in the firmware directory, for instance.

Do we do that in this branch? What do you think?

Comment (dehanj) [reply doc/system_description/software.md:258]: Great, I agree. Documentation close to the source makes it easier for the developer and the one trying to find it. 

No, let's do it in multiple smaller branches and PRs. Makes it easier than one big.

**Review by dehanj — APPROVED** [review]

**Pull request merged**